messages.py

```diff
@@ -0,0 +1,57 @@
+from dataclasses import dataclass
+from datetime import datetime, timedelta, timezone
+
+import constants
+
+# Messages defined here:
+# Submit                                              : /submit <code> <day> <part>
+# Help:                                               : /help
+# Info:                                               : /info
+# Leaderboard                                         : /best <day> <part>
+
+@dataclass
+class SubmitMessage():
```

Comment:
IIRC, dataclasses aren't _supposed_ to have methods, but I've committed that crime too. Let's see how the other refactors go, and then revisit this after release.

Reply:
I don't think there are any rules like that about data-classes. Data-classes basically just add a few convenience methods like `__str__` but there is no expectation that you cannot further add your own methods. There are some examples of this in the [docs](https://docs.python.org/3/library/dataclasses.html) as well.

Reply:
You might find this interesting: https://realpython.com/python-data-classes/